### PR TITLE
Added missed warnings for deprecated schema

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue #2794 

@davidfowl @pranavkm @troydai

### CODE REVIEW

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.DotNet.ProjectModel/ProjectReader.cs [hunk 1]
```diff
@@ -766,5 +767,52 @@ private static bool HasProjectFile(string path)
...
+            string option,
+            string message)
+        {
+            var lineInfo = rawProject.Value<IJsonLineInfo>(option);
+            if (lineInfo == null)
+            {
+                return;
+            }
+
+            project.Diagnostics?.Add(
```
Comment (troydai) [inline src/Microsoft.DotNet.ProjectModel/ProjectReader.cs]: when will this be `null`?

**Comment by troydai** [comment]
:shipit:

**Comment by troydai** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

failure here is transient

**Comment by ajaybhargavb** [comment]
Updated and rebased.

**Pull request merged**